### PR TITLE
App state banner tweaks

### PR DESCRIPTION
**Pull request by ivaniskandar**
* Move download indexing notification to this banner group
* Animate state changes

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/data/notification/Notifications.kt [hunk 1]
```diff
@@ -40,8 +40,6 @@ object Notifications {
     const val ID_DOWNLOAD_CHAPTER_PROGRESS = -201
     const val CHANNEL_DOWNLOADER_ERROR = "downloader_error_channel"
     const val ID_DOWNLOAD_CHAPTER_ERROR = -202
-    const val CHANNEL_DOWNLOADER_CACHE = "downloader_cache_renewal"
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/notification/Notifications.kt:43]: Add it to `deprecatedChannels` so it gets deleted.

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt [hunk 2]
```diff
@@ -66,6 +64,8 @@ class DownloadCache(
      */
     private var lastRenew = 0L
     private var renewalJob: Job? = null
+    val isRenewing: Boolean
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt:67]: Why not just make this a flow? Feels weird to check this based on every cache change in the activity.

File: app/src/main/java/eu/kanade/tachiyomi/data/notification/Notifications.kt (hunk 1 above)
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/data/notification/Notifications.kt:43]: Done

File: app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt (hunk 2 above)
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt]: Done. Still piggybacking off of the `changes` flow but added `distinctUntilChanged()`

**Pull request merged**